e2e_tests/tests/experiment/experiment.py

```diff
@@ -72,6 +72,76 @@ def cancel_trial(trial_id: int) -> None:
     wait_for_trial_state(trial_id, determinedexperimentv1State.STATE_CANCELED)
 
 
+def wait_for_experiment_by_name_is_active(
+    experiment_name: str,
+    min_trials: int = 1,
+    max_wait_secs: int = conf.DEFAULT_MAX_WAIT_SECS,
+    log_every: int = 60,
+) -> int:
+    for seconds_waited in range(max_wait_secs):
+        try:
+            response = bindings.get_GetExperiments(
+                determined_test_session(), name=experiment_name
+            ).experiments
+            if len(response) == 0:
+                time.sleep(1)
+                continue
+            if len(response) > 1:
+                pytest.fail(
+                    f"Multiple experiments with name={experiment_name}, "
+                    f"expected only 1 experiment."
+                )
+            experiment = response[0]
+            experiment_id = experiment.id
+        # Ignore network errors while polling for experiment state to avoid a
+        # single network flake to cause a test suite failure. If the master is
+        # unreachable multiple times, this test will fail after max_wait_secs.
+        except api.errors.MasterNotFoundException:
+            logging.warning(
+                "Network failure ignored when polling for state of "
+                "experiment {}".format(experiment_name)
+            )
+            time.sleep(1)
+            continue
+        except api.errors.NotFoundException:
+            logging.warning(
+                "Experiment not yet available to check state: "
+                "experiment {}".format(experiment_name)
+            )
+            time.sleep(0.25)
+            continue
+
+        if experiment.state == determinedexperimentv1State.STATE_ACTIVE:
+            if experiment.numTrials > min_trials:
+                return experiment_id
+            time.sleep(0.25)
+            continue
+
+        if is_terminal_state(experiment.state):
+            report_failed_experiment(experiment_id)
+
+            pytest.fail(
+                f"Experiment {experiment_id} terminated in {experiment.state.value} state, "
+                f"expected {determinedexperimentv1State.STATE_ACTIVE}"
+            )
+
+        if seconds_waited > 0 and seconds_waited % log_every == 0:
+            print(
```

Comment:
Q: any specific reason you prefer `print` to `log.info`?

Reply:
No. This method is a copy-paste of `wait_for_experiment_state` logic with adjustments to find experiment by name instead of id. I haven't looked into logging, only fixed things necessary for my use case. 

I am not sure if there is any difference between print and log.info for tests infrastructure (e.g., different output?), so I'd vote to keep it as is.